src/hosted-buttons/utils.js

```diff
@@ -289,41 +327,96 @@ export function getButtonColor(
   }
 
-  const buttonContainer = document.createElement("div");
+  buttonContainer.style.cssText = `display: flex; flex-wrap: nowrap; gap: 16px; max-width: 750px; flex-direction: ${flexDirection}`;
+};
 
-  buttonContainer.setAttribute(
-    "style",
-    `display: flex; flex-wrap: nowrap; gap: 16px; max-width: 750px; flex-direction: ${flexDirection}`
+export const getDefaultButton = ({
+  buttonPreferences,
+  eligibleFundingMethods,
+}: HostedButtonPreferences): ?string => {
+  // Return first eligible funding method that is not specified in button preferences
+  return eligibleFundingMethods.find(
+    (fundingMethod) => !buttonPreferences.includes(fundingMethod)
   );
+};
 
-  const primaryButton = document.createElement("div");
-  primaryButton.setAttribute("id", `ncp-primary-button`);
-  primaryButton.setAttribute("style", "flex-grow: 1");
+export const renderStandaloneButton = ({
+  fundingSource,
+  preferences,
+  buttonContainerId,
+  buttonOptions,
+}: RenderStandaloneButtonProps): ZalgoPromise<void> | void => {
+  const Buttons = getButtonsComponent();
 
-  const secondaryButton = document.createElement("div");
-  secondaryButton.setAttribute("id", `ncp-secondary-button`);
-  secondaryButton.setAttribute("style", "flex-grow: 1");
+  const { buttonPreferences, eligibleFundingMethods } = preferences;
+  const { style } = buttonOptions;
 
-  buttonContainer.appendChild(primaryButton);
-  buttonContainer.appendChild(secondaryButton);
+  const isDefault = fundingSource === "default";
 
-  elm?.appendChild(buttonContainer);
-}
+  const fundingSourceName = isDefault
+    ? getDefaultButton({ buttonPreferences, eligibleFundingMethods })
+    : fundingSource;
+
+  // If getDefaultButton doesn't return a button, just return void
+  if (!fundingSourceName) {
+    return;
+  }
+
+  // $FlowFixMe
+  const standaloneButton = Buttons({
+    ...buttonOptions,
+    fundingSource: fundingSourceName,
+    style: {
+      ...style,
+      // $FlowFixMe
+      color: getButtonColor(style.color, fundingSourceName),
+    },
+  });
+
+  if (standaloneButton.isEligible()) {
+    return standaloneButton.render(`#${buttonContainerId}`);
+  }
+
+  getLogger().error(`ncps_standalone_${fundingSourceName}_ineligible`);
+
+  // If the funding source is explicitly defined, but is not eligible,
+  // don't render anything in it's place
+  if (!isDefault) {
+    return;
+  }
+
+  // Filter out ineligible funding method and everything before
+  const indexOfFundingSource =
+    eligibleFundingMethods.indexOf(fundingSourceName);
+  const filteredEligibleFundingMethods = eligibleFundingMethods.filter(
+    (_, index) => index > indexOfFundingSource
+  );
+
+  // If no eligible funding methods remain, return nothing
+  if (!filteredEligibleFundingMethods.length) {
+    return;
+  }
+
+  // Recursively call renderStandaloneButton to attempt to render the next eligible button.
+  return renderStandaloneButton({
```

Comment:
I'm definitely down to make this recursive call happen! There are a few changes that I think would help make the function easier to follow.

1. I would abstract all the logic around handling the `default` button into a separate function. This way it is very clean what logic is used to render the standalone button and what is used to get an eligible `default` button
3. I think the way we're receiving `buttonPreferences` & `eligibleFundingMethods` makes finding an eligible button a bit tricky. Id be open to changing the shape, value or combing them into something that is easier to loop over. I see 2 options but not attached to either and open to others too! 
    - removing values in `eligibleFundingMethods` that appear in `buttonPreferences` so don't have to worry about rendering the paypal button as the `default` button
    - Combining into one array of possible buttons and having a way to only render 2 eligible buttons
3. I'm hoping by making some of these changes we can reduce the number of return statements to make the recursive stuff work!

Hopefully this block of text makes sense and I'm down to pair on it as well. Excelent work so far!

Reply:
I'd love to pair on this next week! 

I tried a couple things here, but since I have to check `Buttons().isEligible()` for each potential button, I landed on this solution.  Would love to think through some other options though! 💯